tsconfig.json

```diff
@@ -0,0 +1,43 @@
+{
+  "compilerOptions": {
```

Comment:
Can you also add `"preserveConstEnums": true,`

Reply:
It looks that this will affect performance...

Reply:
I think the "we won't read" part is not true.  For developers, probably.  For users writing their own triggers, they will probably look at the generated javascript code.

Reply:
In fact I think user won't read nether `raidboss_data.bundle.js` or the trigger file compiled by tsc(both unreadable even if we set this option to true), They should read the original TS file instead.

This is what generated code would look like, and there it's not worth reading.
![image](https://user-images.githubusercontent.com/13553903/107865450-677d7f80-6ea1-11eb-8318-f89bd42beabd.png)

Reply:
They certainly won't read the bundle, but they will likely read typescript -> javascript converted trigger files.  A lot of users copy and paste triggers and so I think there needs to be some solution for them.  I think when publishing to gh-pages (or in a release??) we will need to convert the typescript trigger files to javascript so users have something to copy and paste from in their user trigger files, e.g. [this workflow](https://github.com/quisquous/cactbot/blob/main/docs/CactbotCustomization.md#example-1-changing-the-output-text)

Reply:
I think we can do that with cli flags when generating a preview(maybe, I'm not sure how to call it), and disable it when building the final bundle file.

Reply:
Ok, that works for me, thanks!